src/cosim/osp_config_parser.cpp

```diff
@@ -926,8 +933,16 @@ osp_config load_osp_config(
         throw std::invalid_argument(oss.str());
     }
 
+    if (simInfo.startTime > simInfo.endTime) {
+        std::ostringstream oss;
```

Comment:
It looks to me like this error handler will be triggered if an end time is not specified in the configuration file, because then `simInfo.endTime` will still be zero. Or am I misreading this?

Reply:
You are right of course :facepalm: 

Seems making `endTime ` `std::optional` everywhere is the way to go here then. I also added a separate `config_end_time_test` to test a system structure with/without `endTime` configured.